### ISSUE TITLE
Fix the api for projects to include project users

### ISSUE THREAD

**Issue by jbubar** [hackforla/VRMS#1163]
### Blocks
- #1152 

### Overview
The projects api does not currently return a list of users managing the project. This fix requires updating the projects endpoint to return users assigned to it: users.managedProjects (the managedByUsers field in the projects table does not appear to ever contain any user ids).

### Action Items
The following action items have already been completed via other PRs (see [comment](https://github.com/hackforla/VRMS/issues/1163#issuecomment-4076592977) for updates)
- [x] research how to create a many to many relationship with mongo
- [x] do some research about running a 'migration' to backfill all the data. (this will be a separate issue)
- [x] rewrite the model to connect the projects to the managedByUsers (add the foreign key to the projects model)
- [x] Create an issue to run a 'Migration' to have update all of the old data
- [ ] update the api endpoint to return user's info (user name, id, and email)
- [ ] comment on the  #1052 issue to unblock it

### Resources/Instructions
[Issue where this was discovered](https://github.com/hackforla/VRMS/issues/1152#issuecomment-1175872747)

**Comment by jbubar** [comment]
I want to encourage @chukalicious to do this issue. She is super smart and I feel like this would be a good opportunity to learn some backend.. I do need a bit of help breaking it down tho

**Comment by JackHaeg** [comment]
Action items need to be re-written. This is an old issue, however, this will allow for the website team to know who is associated with the project when updates are done on VRMS.

<img width="967" alt="Screenshot 2023-10-19 at 6 58 54 PM" src="https://github.com/hackforla/VRMS/assets/134463646/30ae9e47-7a9c-4cae-839d-0773a8829bfc">

**Comment by JackHaeg** [comment]
# Update:
:white_check_mark: Item 1: Research MongoDB many-to-many relationship
  - Status: COMPLETE
  - Evidence: The managedByUsers field exists in the Project model (line 36 of project.model.js) as managedByUsers: [{
  type: String }], and the corresponding managedProjects field exists in User model (line 39 of user.model.js)
  
:white_check_mark: Item 2: Research migration to backfill data
  - Status: COMPLETE
  - Evidence: Issue #1386 was created specifically for this: "Database migration to include project managers on project db
   entry" (CLOSED, assigned to jng34)
  
:white_check_mark: Item 3: Rewrite model to connect projects to managedByUsers
  - Status: COMPLETE
  - Evidence:
    - Project model has the field defined
    - Controller has updateManagedByUsers() (line 70-104) that handles add/remove operations
    - Controller has bulkUpdateManagedByUsers() (line 106-131) for bulk operations
    - Routes defined for PATCH /projects/:ProjectId and POST /projects/bulk-updates
  
:white_check_mark: Item 4: Create a migration issue
  - Status: COMPLETE
  - Evidence: Issue #1386 exists and is closed
  ---
  
:warning:  Note: The relationship is implemented as simple string arrays rather than MongoDB ObjectId references. Also, the API endpoints currently return user IDs only, not full user info (name, email) as required by action item 5.

# Next steps: 
- Work should begin on item 5

**Comment by pillarijake** [comment]
Hi @juliagab56 let me know if you're around at some point to discuss this request from Bonnie. Thanks

Need to be able to make someone a PM and remove access (looking for solution that is the lowest lift)

Options:

1. Log into VRMS, go to Users, click User Search, Results by Name, type in User, next to project there should be a toggle for Manager (preferred solution)
2. Interstitial would appear to make user a PM when they click add Project.
